lib/lita/handlers/github_prs/git_diff.rb

```diff
@@ -0,0 +1,40 @@
+# frozen_string_literal: true
+
+module Lita
+  module Handlers
+    module GithubPrs
+      class GitDiff
+        def initialize(diff)
+          @diff = diff
+        end
+
+        def modified_files
+          @diff
+            .files
+            .select { |file| file.status == 'modified' }
+            .map(&:filename)
+        end
+
+        def added_files
+          @diff
+            .files
+            .select { |file| file.status == 'added' }
+            .map(&:filename)
+        end
+
+        def pull_requests
+          @diff
+            .commits
+            .map(&:commit)
+            .select { |commit| pull_request_commit?(commit) }
```

Comment:
Depending on your preference, another option:

```ruby
.select(&method(:pull_request_commit?))
```

Reply:
Interesting, didn't know about `&method`. Like it!

Reply:
:+1: For more background, really `method` just gets a method off an object and gives you a `Method` object which happens to respond to `call` (which is what lets you put `&` in front).  One of the most useful things I ever learned in ruby was that you can do `method(:method_name).source_location` and it tells you the file / line where that is defined.  Especially useful when it might be dispatching to multiple places.

Reply:
I really hope most of the code I will look at in the future will have good transparency or be very expressive so that I don't have to use this but I suppose it'll be useful when many modules are included somewhere! 👍

Reply:
For sure, though when you're using gems it's also a great way to know exactly where to look in the gem to read the code